apps/api-gateway/src/board/board.controller.ts

```diff
@@ -62,7 +62,7 @@ export class BoardController {
       is_open: true,
     },
   })
-  public async getPostById(@Param('id') id: string) {
+  public async getPostById(@Param('id', ParseIntPipe) id: number) {
     return await this.boardService.getPostById(id);
   }
```

Comment:
```suggestion
  public async getPostById(@Param('id') id: number) {
    return await this.boardService.getPostById(id);
  }
```

현재 api-gateway 측 main.ts 의 app 설정 부분에
```typescript
 app.useGlobalPipes(
    new ValidationPipe({
      transform: true,
      whitelist: true,
    }),
  );
```
위와 같이 설정이 되어 있는데요!
transform option을 true 로 할당 해놨기 때문에,
해당 코드내에서 ParseIntPipe까지 사용할 이유는 없을 것 같습니다.

아래 공식문서 한번 참고해 보세요!
[NestJS: Validation -Transform Paylaod Objects](https://docs.nestjs.com/techniques/validation#transform-payload-objects)

Reply:
변환 옵션 설정 부분을 확인하지 못했던 것 같습니다. 
공식 문서도 첨부해 주셔서 이해가 수월했습니다. 감사합니다!
수정된 커밋입니다 : 5ae6b9a50b02c5db9e08e0f31f619b193d504e2c